src/thunderbots/software/geom/circle.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <functional>
+
+#include "geom/point.h"
+
+class Circle final
+{
+   public:
```

Comment:
Would `private` be better since I guess we don't want direct access to the `origin` and `radius`? Similar to the rest of the classes.